gsplat/cuda/csrc/helpers.cuh

```diff
@@ -104,36 +110,26 @@ inline __device__ float4 transform_4x4(const float *mat, const float3 p) {
 }
 
 inline __device__ float2 project_pix(
-    const float *mat, const float3 p, const dim3 img_size, const float2 pp
+    const float2 fxfy, const float3 p_view, const float2 pp
 ) {
-    // ROW MAJOR mat
-    float4 p_hom = transform_4x4(mat, p);
-    float rw = 1.f / (p_hom.w + 1e-6f);
-    float3 p_proj = {p_hom.x * rw, p_hom.y * rw, p_hom.z * rw};
-    return {
-        ndc2pix(p_proj.x, img_size.x, pp.x), ndc2pix(p_proj.y, img_size.y, pp.y)
-    };
+    float rw = 1.f / (p_view.z + 1e-6f);
+    float2 p_proj = { p_view.x * rw, p_view.y * rw };
+    // TODO: check if +-0.5 should be added to pix coords or not
+    // this depends on rasterize
+    float2 p_pix = { p_proj.x * fxfy.x + pp.x, p_proj.y * fxfy.y + pp.y };
+    return p_pix;
```

Comment:
This may slightly change the results: the current code subtracts `-0.5f` from the pixel coordinates, which is not done here. It can be added back but is there a good reason to do so? If fx, fy ppx, ppy originate from e.g., COLMAP, the pinhole projection formula does not include such a term.

Reply:
@oseiskar I think we really need to add this back. Otherwise it would bring quality degradation in splatfacto. For an image, each pixel's index (i,j) actually represents the area of intensity whose center 2d point is at location (i+0.5, j+0.5).

Reply:
Commented there https://github.com/nerfstudio-project/gsplat/pull/151#issuecomment-2027978449